Client/Frontend/Sync/BraveCore/BraveServiceStateObserver.swift

```diff
@@ -0,0 +1,25 @@
+// Copyright 2021 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+import Foundation
+
+class BraveServiceStateObserver: NSObject {
+    
+    // MARK: Static
+    
+    static let coreServiceLoadedNotification = "BraveServiceStateDidLoaded"
+    
+    static var isServiceLoadStatePosted = false
+
+    // MARK: Private
+    
+    func postServiceLoadedNotification() {
+        NotificationCenter.default.post(
+            name: Notification.Name(rawValue: BraveServiceStateObserver.coreServiceLoadedNotification),
+            object: nil)
+        
+        BraveServiceStateObserver.isServiceLoadStatePosted = true
+    }
```

Comment:
Because to enableDifferent Sync Types we have to be sure brave-core service is loaded and ready to be used. We add an observer (BraveServiceStateObserver.coreServiceLoadedNotification) to enable User Selected Types For Sync. And this can happen either bookmarksAPI is loaded or historyAPI receives isBackEndloaded. So both `HistoryServiceStateObserver` and `BookmarkModelStateObserver` are BraveServiceStateObservers.

But you are right this can be guarded before posting so Fix at https://github.com/brave/brave-ios/pull/3703